dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -432,6 +432,426 @@ bool TypeLookupService::complete_to_minimal_type_object(const TypeObject& cto, T
   }
 }
 
+void TypeLookupService::insert_dynamic_member(DynamicType_rch& dt, const DynamicTypeMember_rch& dtm)
+{
+  dt->member_by_index.push_back(dtm);
+  if (dtm->descriptor_->id != MEMBER_ID_INVALID) {
+    dt->member_by_id.insert(std::make_pair(dtm->descriptor_->id , dtm));
+  }
+  dt->member_by_name.insert(std::make_pair(dtm->descriptor_->name , dtm));
+}
+
+void TypeLookupService::handle_tryconstruct_flags(MemberDescriptor*& md, MemberFlag mf)
+{
+  if (mf & (1 << 0)) {
```

Comment:
Instead of having expressions like `(1 << 0)` in the implementation, either use a named constant or just the plain integer literal that you'd get from evaluating that expression.